runtime/auth/aws-signing-common/common/src/aws/smithy/kotlin/runtime/auth/awssigning/AbstractAwsChunkedByteReadChannel.kt

```diff
@@ -203,4 +204,48 @@ public abstract class AbstractAwsChunkedByteReadChannel(
         chunkOffset = 0
         return trailerBody
     }
+
+    /**
+     * Given a signing config, make a deep copy, changing the signatureType and hashSpecification configuration values
+     * to specify chunk signing.
+     * @param config the original signing config
+     * @return an [AwsSigningConfig] which can be used by a signer to sign chunks
+     */
+    private fun getChunkSigningConfig(config: AwsSigningConfig): AwsSigningConfig = AwsSigningConfig {
+        region = config.region
+        service = config.service
+        signingDate = config.signingDate
+        shouldSignHeader = config.shouldSignHeader
+        algorithm = config.algorithm
+        signatureType = AwsSignatureType.HTTP_REQUEST_CHUNK // signature is for a chunk
+        useDoubleUriEncode = config.useDoubleUriEncode
+        normalizeUriPath = config.normalizeUriPath
+        omitSessionToken = config.omitSessionToken
+        hashSpecification = HashSpecification.CalculateFromPayload // calculate the hash from the chunk payload
+        signedBodyHeader = config.signedBodyHeader
+        credentialsProvider = config.credentialsProvider
+        expiresAfter = config.expiresAfter
+    }
```

Comment:
**Nit**: Could be shortened with `toBuilder()`:

```kotlin
private fun getChunkSigningConfig(config: AwsSigningConfig): AwsSigningConfig = config
    .toBuilder()
    .apply {
        signatureType = AwsSignatureType.HTTP_REQUEST_CHUNK // signature is for a chunk
        hashSpecification = HashSpecification.CalculateFromPayload // calculate the hash from the chunk payload
    }
    .build()
```

Applies to `getTrailingHeadersSigningConfig` as well.